adbe/adb_enhanced.py

```diff
@@ -2090,6 +2090,22 @@ def toggle_location(turn_on):
     execute_adb_shell_settings_command(cmd)
 
 
+@ensure_package_exists
+def set_debug_app(app_name, wait_for_debugger, persistent):
```

Comment:
Do you know in which Android API version was `am set-debug-app` introduced?

I use `_error_if_min_version_less_than(<version>)` to error out if the command is going to fail anyways.

Reply:
As I can see in sources, `set-debug-app` and `clear-debug-app` have been added in year of 2011: https://cs.android.com/android/_/android/platform/frameworks/base/+/d8c98fee02b5498270344727d1289004d98c0e73
Unfortunately, I have no idea how to find the actual Android version in specific commit: https://cs.android.com/android/_/android/platform/frameworks/base/+/d8c98fee02b5498270344727d1289004d98c0e73:. Please, let me know, if you know how to find it.
As I can see here: https://cs.android.com/android/_/android/platform/frameworks/base/+/d8c98fee02b5498270344727d1289004d98c0e73:api/current.txt;l=14409, the actual API version is 15. Unfortunately, I can't find emulator system images of API 14 or 15 on arm64 to check that.

Reply:
No worries, these are old enough that practically any device one would test will support them.